### DISCUSSION TITLE
Access async action id inside of action

### DISCUSSION THREAD

**Discussion by talgat-ruby** [hasura/graphql-engine#8750] [Question]
Hi

For example, lets say I have async action `someAsyncAction`. I can access `action_id` in the **client** via graphql.
```graphql
mutation SomeAsyncActionRequest($name: String!) {
  someAsyncAction(name: $name)
}
```
```json
{
  "data": {
    "someAsyncAction": "23b1c256-7aff-4b95-95bd-68220d9f93f2"
  }
}
```

But this is **client**. How about within action code?  I want to save the `action_id` ro db to show its status only for a particular user.
```js
module.exports = async (req, res) => {
  const { input, action } = req.body // can access action description: name, but why not id?
  const { name } = input

  try {
    const resp = await heavyRequest(name)

    res.send({
      success: resp.data.success
    })
  } catch (err) {
    res.status(400).json({
      message: err.message
    })
  }
}
```

The only workaround I found is to Proxy asynchronous action with synchronous action. Isn't it overhead.

**Comment by adas98012** [comment]
You can get the action id of an async action already, via the mutation call. Here is the document link:

https://hasura.io/docs/latest/actions/async-actions/

Let us know if the info suffice.

**Reply by talgat-ruby** [reply]
@adas98012 Thanks, but I know about that. But what about within action itself. Please refer to question I've updated.

**Comment by adas98012** [comment]
@talgat-ruby I have opened a new [issue](https://github.com/hasura/graphql-engine/issues/8903) in Github as an enhancement request. I have notified our engineering team and you should be able to track the status of the issue as well.

**Reply by talgat-ruby** [reply]
Thank you @adas98012
